src/twisted/internet/task.py

```diff
@@ -20,11 +31,15 @@
 from twisted.python.failure import Failure
 from incremental import Version
 
-from twisted.internet import base, defer
-from twisted.internet.interfaces import IReactorTime
+from twisted.internet.base import DelayedCall
+from twisted.internet.defer import Deferred, ensureDeferred, maybeDeferred
+from twisted.internet.interfaces import IDelayedCall, IReactorCore, IReactorTime
 from twisted.internet.error import ReactorNotRunning
 
 
+_T = TypeVar("_T")
```

Comment:
A good example of the difference, which bit me a few times before I understood it, is the [AnyStr](https://docs.python.org/3/library/typing.html#typing.AnyStr) type var, which is defined like your type `A`. Consider these two functions:

```python

def foo(arg: AnyStr) -> AnyStr:
  ...

def bar(arg: Union[str, bytes]) -> Union[str, bytes]:
  ...
```

Both functions can take either str or bytes and can return either str or bytes. However, if you call `foo` with a str, it is constrained to return a str, not bytes — for the scope of that call, `AnyStr` is bound specifically to str. On the other hand, it's perfectly valid for `bar` to return bytes if you pass it a str, or even to return a random choice between strs and byteses at runtime. `bar`'s arg type and return type aren't connected to each other (at least, not connected by the type annotations).

Another, more subtle, difference, is that `AnyStr` can't be bound to `Union[str, bytes]`! So if a *caller* has some value that could be str or bytes but it doesn't know which (i.e., the caller's variable has type `Union[str, bytes]`) it can't call `foo()`. I ran into that situation [here](https://github.com/twisted/twisted/pull/1576/commits/8f483dc7e6ef0340ec86a9ee950185a537c54671).

Reply:
`AnyStr` usage is clear ... but what is the purpose of `_T = TypeVar("_T")` ?

Does it mean that with this `foo` accepts any type, but arg1 and arg2 and return must be of same type   ?

```python
def foo(arg1: _T, arg2: _T) -> _T:
    ...
```

Maybe rename it to `_SameType = TypeVar('_SameType')

Reply:
> Does it mean that with this `foo` accepts any type, but arg1 and arg2 and return must be of same type ?

Yes exactly.

I think it's reasonable keeping the name since `T` and `U` are pretty much universally used for generic type variables like these. A name like `SameType` would be confusing because it's redundant; *all* type variables are the same type as themselves, so I'd go looking for what *else* `SameType` was the same type as.

If the variable is used over many lines or is constrained in some interesting way, etc., then I think it's worth giving it a descriptive name. I've been using `_TSomething` but there might be a better convention out there.

Reply:
`_T` is used all other the place in type hints, so I'd rather not depart from that convention, which is also in use Twisted elsewhere.

If Twisted want to adopt a different convention, that's fine, but this isn't the PR for that.  Perhaps a discussion, then edit to contributor docs, then fix the existing use cases.